app/api/route.js

```diff
@@ -0,0 +1,298 @@
+import Author from "@/models/authors";
+import { ApolloServer } from "@apollo/server";
+import { startServerAndCreateNextHandler } from "@as-integrations/next";
+import DataLoader from "dataloader";
+import { gql } from "graphql-tag";
+import moment from "moment/moment";
+import { validateImageUrl } from "../utils/imageValidator";
+const Book = require("../../models/books");
+const { Op } = require("sequelize");
+
+const typeDefs = gql`
+  type Book {
+    id: Int!
+    title: String!
+    description: String!
+    publishedAt: String!
+    profilePhotoUrl: String
+    author: Author!
+    author_id: Int!
+    reviews: [Review]
+  }
+
+  type Review {
+    rating: Int!
+    comment: String
+  }
+
+  type Author {
+    id: Int!
+    name: String!
+    biography: String!
+    profilePhotoUrl: String
+    books: [Book]
+    date_of_birth: String!
+  }
+
+  type BookConnection {
+    books: [Book!]!
+    totalCount: Int!
+    hasMore: Boolean!
+  }
+
+  type AuthorConnection {
+    authors: [Author!]!
+    totalCount: Int!
+    hasMore: Boolean!
+  }
+
+  type Query {
+    book(id: Int!): Book
+    books(page: Int, size: Int, searchTitle: String): BookConnection!
```

Comment:
as this gets more complex, try to use `input AddBook` something like this to have reusable inputs as well.